### ISSUE TITLE
Sound issues Crash Team Racing: Nitro-Fueled

### ISSUE THREAD

**Issue by teamracingfan** [yuzu-emu/yuzu#5189]
hi team

I notice sound issues from version YuzuEA-1123 (same issues for all subsequent versions I tried, YuzuEA-1160 YuzuEA-1175 and now YuzuEA-1222). The sound for me is get muffled or cut out during a race. 
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/5682961/yuzu_log.txt)

I don't have this sound issue with 1122 (the last working version for me) and earlier.

**Comment by titan-91** [comment]
This happens to me as well about 20% of the time. For me though, the sound appears to be lagging (frequently cutting in and out) but the game is running full speed.

Linux log attached.
[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/5686064/yuzu_log.txt)

**Comment by teamracingfan** [comment]
this is a sample, the good audio version with YuzuEA-1122 (and all earlier versions)
https://user-images.githubusercontent.com/71040364/102712631-f4052a80-42c2-11eb-9eaf-0525fdb6f185.mp4

and the bad version from YuzuEA-1123 (and subsequent)
https://user-images.githubusercontent.com/71040364/102712634-fb2c3880-42c2-11eb-82cc-1da4869f5e55.mp4

**Comment by teamracingfan** [comment]
what's changed from 1122 to 1123 and is possible fix it ?

**Comment by Krnlsoft** [comment]
Hi,
the following things changed:

![image](https://user-images.githubusercontent.com/6281198/102712751-b785fe80-42c3-11eb-93be-36d52b19de2d.png)

**Comment by teamracingfan** [comment]
thanks Krnlsoft for your quickly reply

ok so I assume thing changed to fix bugs with other games...

but now audio in Crash Team Racing: Nitro-Fueled is broken.

**Comment by Krnlsoft** [comment]
The issue could be in the audio_renderer.cpp file, but Im no expert, just trying to give some hints:

[audio_renderer.diff.txt](https://github.com/yuzu-emu/yuzu/files/5720893/audio_renderer.diff.txt)

**Comment by LuismaSP89** [comment]
Same happens in Monster Boy, some builds ago the audio worked flawlessly, in the recent builds it gots a lot of stuttering.

**Comment by cfurgang** [comment]
This is also happening for me. Reverting to EA-1122 fixed all audio issues.

**Comment by Krnlsoft** [comment]
Hi, the following change solved the issues (at least for me):

https://github.com/Krnlsoft/yuzu/commit/871729ec172cf9f25eaaecf57c4d341bff8d758f

Don't ask me why, cause I don't know much, I only reverted some lines of code, compiled and tested.

**Comment by teamracingfan** [comment]
> 
> 
> Hi, the following change solved the issues (at least for me):
> 
> [Krnlsoft@871729e](https://github.com/Krnlsoft/yuzu/commit/871729ec172cf9f25eaaecf57c4d341bff8d758f)
> 
> Don't ask me why, cause I don't know much, I only reverted some lines of code, compiled and tested.

and how can I use this change ? in YuzuEA-1289 doesn't work, same issue

**Comment by Krnlsoft** [comment]
> > Hi, the following change solved the issues (at least for me):
> > [Krnlsoft@871729e](https://github.com/Krnlsoft/yuzu/commit/871729ec172cf9f25eaaecf57c4d341bff8d758f)
> > Don't ask me why, cause I don't know much, I only reverted some lines of code, compiled and tested.
> 
> and how can I use this change ? in YuzuEA-1289 doesn't work, same issue

You have to apply the patch and compile from source code

**Comment by borismul** [comment]
@Krnlsoft It didn't work for me. The audio is good for a while but after a while the background music becomes very fast, while the engine sounds are delayed. Very weird. Before the patch the sounds were too slow as you mentioned. It usually transitioned from normal to the weird sound after loading screens. With the patch in transitioned during a race.

**Comment by titan-91** [comment]
I dare say this may be fixed for me. I played the Pinstripe boss race twice in a row, closing and opening Yuzu 2 different times, with Vulkan and OpenGL. Not once did I have the audio cracking bug. Fingers crossed some other revision resolved it. This is using the new mainline build from 13 hours ago (today).

**Comment by borismul** [comment]
In CTR I have the feeling the audio transitioning from good to bad might be related to the performance of the PC. I noticed that when I unplug my laptop it happens much faster. Also the laptop is not super fast. Intel i7 6 cores from 2 years ago (I don't remember the exact CPU name) and a GTX 1060M.

**Comment by Krnlsoft** [comment]
Test mainline build (5b1e158e6) without my patch, but the issue still persist.
It's indeed related to CPU performances, mine is a really old one (Intel® Core™2 Quad Q9300).

**Comment by teamracingfan** [comment]
> 
> 
> > > Hi, the following change solved the issues (at least for me):
> > > [Krnlsoft@871729e](https://github.com/Krnlsoft/yuzu/commit/871729ec172cf9f25eaaecf57c4d341bff8d758f)
> > > Don't ask me why, cause I don't know much, I only reverted some lines of code, compiled and tested.
> > 
> > 
> > and how can I use this change ? in YuzuEA-1289 doesn't work, same issue
> 
> You have to apply the patch and compile from source code

I don't know how to do it.
Why all new YuzuEA releases can't be fixed ?

**Comment by teamracingfan** [comment]
@Krnlsoft  Could you please attach here your patched yuzu version ?

**Comment by Krnlsoft** [comment]
> > > > Hi, the following change solved the issues (at least for me):
> > > > [Krnlsoft@871729e](https://github.com/Krnlsoft/yuzu/commit/871729ec172cf9f25eaaecf57c4d341bff8d758f)
> > > > Don't ask me why, cause I don't know much, I only reverted some lines of code, compiled and tested.
> > > 
> > > 
> > > and how can I use this change ? in YuzuEA-1289 doesn't work, same issue
> > 
> > 
> > You have to apply the patch and compile from source code
> 
> I don't know how to do it.
> Why all new YuzuEA releases can't be fixed ?

Can't tell... I think the team is busy with other issues.
I could open a pull request, but my fix is not a valid pull request, it's just a hack / workaround.


> @Krnlsoft Could you please attach here your patched yuzu version ?

I compile only for linux, I could compile it for windows but I don't have the time nor the will to do it.

It's better to wait the devs to take a look at this thread.

**Comment by titan-91** [comment]
I would definitely be interested in a Linux binary or appimage with the lath if you could.

**Comment by teamracingfan** [comment]
> It's indeed related to CPU performances, mine is a really old one (Intel® Core™2 Quad Q9300).

I have an Intel® Core™ i3-9100 Processor 
https://ark.intel.com/content/www/us/en/ark/products/134870/intel-core-i3-9100-processor-6m-cache-up-to-4-20-ghz.html
my cpu is from 2019... not i5 or i7 but one of the best i3...

I am able to play with version yuzu EA 1122, 60 fps... sound and graphic... almost everything, game functions flawless with no audio or graphical glitches, all tested functionality works, except 1 track where I have another [error described here.](https://github.com/yuzu-emu/yuzu/issues/4644). Because of this I wanted use the latest versions..

if audio works with EA 1122 should work with newer version too.

**Comment by Krnlsoft** [comment]
> I would definitely be interested in a Linux binary or appimage with the patch if you could.

I was about to give you a version to test out, but the issues is random.
So my patch is not working.

**Comment by gcasu** [comment]
I confirm that the issue is still there in Nitro Fueled. Graphic goes quite well at 60FPS with OpenGL and a nVidia 960M, but sound gets cracked usually after one race only, sometimes at the main menu.

**Comment by reggermont** [comment]
Confirm the issue. Still present in 2022

**Comment by gcasu** [comment]
Can anyone please verify if this issue would be solved by #8545 ?

**Comment by german77** [comment]
Latest EA is 2836. The one you mentioned doesn't have PR 8545

**Comment by Max-feded** [comment]
the issue still present for me

**Comment by teamracingfan** [comment]
> Latest EA is 2836. The one you mentioned doesn't have PR 8545

Tested 2836 and I have the same, audio is still broken

**Comment by Krnlsoft** [comment]
The audio stutter occur when other processes (ex browser, os etc) use the CPU.
Pause and resume solve the stutter.

**Comment by Max-feded** [comment]
> The audio stutter occur when other processes (ex browser, os etc) use the CPU. Pause and resume solve the stutter.
> 
> yuzu-1141

They fixed the issue, it was an issue on 4 core and 4 thread CPUs, now the game is fine for me

**Comment by Krnlsoft** [comment]
> > The audio stutter occur when other processes (ex browser, os etc) use the CPU. Pause and resume solve the stutter.
> > yuzu-1141
> 
> They fixed the issue, it was an issue on 4 core and 4 thread CPUs, now the game is fine for me

build number? Thanks

**Comment by Max-feded** [comment]
> > > The audio stutter occur when other processes (ex browser, os etc) use the CPU. Pause and resume solve the stutter.
> > > yuzu-1141
> > 
> > 
> > They fixed the issue, it was an issue on 4 core and 4 thread CPUs, now the game is fine for me
> 
> build number? Thanks

Try the latest build, i tried on early access

**Comment by gcasu** [comment]
This issue seems to be resolved for me as expected after #8545 was merged into mainline

**Comment by Max-feded** [comment]
> This issue seems to be resolved for me as expected after #8545 was merged into mainline

Precisely for me the issue was fixed when they implemented a better version of CoreTiming, so i think we can close the issue since the issue is fixed now.

**Comment by teamracingfan** [comment]
> The audio stutter occur when other processes (ex browser, os etc) use the CPU. Pause and resume solve the stutter.
> 
> yuzu-1141

I have still the audio issue but I confirm this workaround works

If I press F4 twice (pause and resume) it resolves the stutter. Thanks for this @Krnlsoft Krnlsoft

The issue now is every 20 minutes I have the stutter and I need to press F4...

Only yuzu is opened not browsers or other software..

**Comment by Krnlsoft** [comment]
> This issue seems to be resolved for me as expected after #8545 was merged into mainline

yuzu-1143 released today seems to have solved the issues for me!

**Comment by Krnlsoft** [comment]
> > The audio stutter occur when other processes (ex browser, os etc) use the CPU. Pause and resume solve the stutter.
> > yuzu-1141
> 
> I have still the audio issue but I confirm this workaround works
> 
> If I press F4 twice (pause and resume) it resolves the stutter. Thanks for this @Krnlsoft Krnlsoft
> 
> The issue now is every 20 minutes I have the stutter and I need to press F4...
> 
> Only yuzu is opened not browsers or other software..

tried yuzu-1143 released today, same issue for me too, but the stutter is less frequent and getting better

**Comment by gcasu** [comment]
> > This issue seems to be resolved for me as expected after #8545 was merged into mainline
> 
> Precisely for me the issue was fixed when they implemented a better version of CoreTiming, so i think we can close the issue since the issue is fixed now.

Do you refer to #8650 ?

**Comment by Max-feded** [comment]
> > > This issue seems to be resolved for me as expected after #8545 was merged into mainline
> > 
> > 
> > Precisely for me the issue was fixed when they implemented a better version of CoreTiming, so i think we can close the issue since the issue is fixed now.
> 
> Do you refer to #8650 ?

No, who created the pull request was Blinkhawk

**Comment by SirvanD** [comment]
The audio issue for Crash Team Racing Nitro-Fueled still persists, 
Tried Latest EA and latest Mainland, but no luck, only workaround pause-replay game works but audio stutter comes back after a few minutes. 
Version 1122 works fine as OP stated.

**Comment by jksprat** [comment]
> The audio issue for Crash Team Racing Nitro-Fueled still persists, Tried Latest EA and latest Mainland, with and without Audio output setting to SDL but no luck, only workaround pause-replay game works but audio stutter comes back after a few minutes. Version 1122 works fine as OP stated.

Confirming this is the case with the latest version as of this posting.

**Comment by MMOStars** [comment]
First time using the emu, wanted to try crash, yeah the audio is crackling in the latest build, overwritten the files with what OP suggested and that fixed the audio issue.

**Comment by SirvanD** [comment]
> First time using the emu, wanted to try crash, yeah the audio is crackling in the latest build, overwritten the files with what OP suggested and that fixed the audio issue.

Can you share the link in OP's post, for some reason it doesn't open for me. So in short, you changed the audio files back to how they were in version 1122?

**Comment by EslamHULK** [comment]
![Screenshot 2023-08-01 185701](https://github.com/yuzu-emu/yuzu/assets/35035933/60a1cb9a-11b6-426a-8f39-7153b1de0dbe)
latest version of YUZU  Vulkan not working in crash team racing
![Screenshot 2023-08-01 185837](https://github.com/yuzu-emu/yuzu/assets/35035933/edf17767-0a1c-4be0-9e08-97de6f1f99a0)
latest version of YUZU  Vulkan not working in crash team racing
   
but in old version look it work
![Screenshot 2023-08-01 010004](https://github.com/yuzu-emu/yuzu/assets/35035933/4107d8d1-d7bc-48b9-8a31-bb4bb01f54af)
![Screenshot 2023-08-01 011442](https://github.com/yuzu-emu/yuzu/assets/35035933/f6e7defd-a419-4b10-a63b-ad07510a6127)

 can fix it soon , and thank u

**Comment by EslamHULK** [comment]
i say old version work in vulkan but new versions vulkan dosent work

**Comment by EslamHULK** [comment]
@mzso  i say old version work in vulkan but new versions vulkan dosent work . look
![Screenshot 2023-08-03 055054](https://github.com/yuzu-emu/yuzu/assets/35035933/0918e437-5290-461b-8227-2b97b1cd95cd)
YUZU 3790

**Comment by mzso** [comment]
@EslamHULK What should I be seeing? All i see is a snapshot of the game.

**Comment by EslamHULK** [comment]
@mzso   so test game > crash team racing by program and check Vulkan in new version of program and see what is happen
![3791](https://github.com/yuzu-emu/yuzu/assets/35035933/402c6025-1ca5-4223-9a56-0c42a73a2c02)
![1516](https://github.com/yuzu-emu/yuzu/assets/35035933/6c40e1ed-4750-490b-b5cf-7cc387cef2ea)
this 2 version 1516 and 3791 , latest versions look the bug or some thing is wrong in program

**Comment by legend800** [comment]
@EslamHULK 
Please stop spamming this thread about audio issues with a launch problem. Go create a new issue.

The game loads fine with VK on the latest version here, so there's something else going on with your files. Take it to Discord to troubleshoot problems on your end not github. 

<img width="815" alt="2023-08-03 22_59_23-Switch" src="https://github.com/yuzu-emu/yuzu/assets/20983844/2a27e1c0-45b7-41e3-b163-2ca4e932cf30">

There's also a lot of good window screenshot apps out there. ;)

**Comment by EslamHULK** [comment]
@legend800 what configuration selected

**Comment by EslamHULK** [comment]
@legend800 

> @EslamHULK Please stop spamming this thread about audio issues with a launch problem. Go create a new issue.
> 
> The game loads fine with VK on the latest version here, so there's something else going on with your files. Take it to Discord to troubleshoot problems on your end not github.
> 
> <img alt="2023-08-03 22_59_23-Switch" width="815" src="https://user-images.githubusercontent.com/20983844/258309407-2a27e1c0-45b7-41e3-b163-2ca4e932cf30.png">
> There's also a lot of good window screenshot apps out there. ;)

this screen shot not first page in the game , u know what i talk about
![Screenshot 2023-08-04 205123](https://github.com/yuzu-emu/yuzu/assets/35035933/832787f4-27cb-4889-ae8f-e2946247c8ab)

**Comment by EslamHULK** [comment]
@yuzu-emu
@pineappleEA
@legend800 
![Screenshot 2023-08-05 185405](https://github.com/yuzu-emu/yuzu/assets/35035933/6ed98140-e842-42f9-986b-29cf632d798c)

![Screenshot 2023-08-05 185318](https://github.com/yuzu-emu/yuzu/assets/35035933/6b8f65a3-3bb0-459a-a886-00c6f93d9498)
![Screenshot 2023-08-05 185227](https://github.com/yuzu-emu/yuzu/assets/35035933/d2520b58-e531-49b0-aa08-10841b696e06)
![Screenshot 2023-08-05 185526](https://github.com/yuzu-emu/yuzu/assets/35035933/bc98535c-7d75-49bf-b9ca-d52da8ddf851)
![Screenshot 2023-08-05 185516](https://github.com/yuzu-emu/yuzu/assets/35035933/967ea39a-2493-490a-8b85-db581545781a)

**Comment by legend800** [comment]
@EslamHULK Do you understand what stop spamming means? What are you even posting above? Makes no sense. 

Stop posting on a thread about AUDIO issues and take it to discord. Learn how to take a screenshot also of the emulator instead of your weird desktop full of videos that you need instant access to for some strange reason.

**Comment by EslamHULK** [comment]
Okay . Sorry about screen shot . But I need to fix this problem about
graphic of switch emulator

في السبت، 5 أغسطس، 2023 6:06 م legend800 ***@***.***> كتب:

**Comment by cjunisl** [comment]
the audio bug still persists in EA3932. Pause and Resume temporarily fixes it until the next loading screen.
